api/src/pcapi/core/subscription/phone_validation/api.py

```diff
@@ -115,6 +116,19 @@ def _check_and_update_phone_validation_attempts(redis: Redis, user: users_models
         redis.expire(phone_validation_attempts_key, settings.PHONE_VALIDATION_ATTEMPTS_TTL)
 
 
+def _send_sms_with_retry(phone_number: str, message: str, max_attempt: int = 3) -> None:
+    for i in range(max_attempt):
+        try:
+            sms_notifications.send_transactional_sms(phone_number, message)
+            return
+        except requests.ExternalAPIException as exception:
+            if exception.is_retryable:
+                if i < max_attempt - 1:
+                    continue
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Pas besoin de "timer" pour laisser du temps entre les retry ici ?
On risque de boucler instantanément ici non ?

Reply:
Bonne question. J'imagine qu'il y aura quand même eu un certain temps après le début de la requête précédente. Ce qui pourrait pallier des pertes réseau. J'invente un peu

Reply:
J'ai mis un time.sleep